test/object-store/migrations.cpp

```diff
@@ -577,29 +577,267 @@ TEST_CASE("migration: Automatic") {
+            realm->update_schema(schema, 1);
+            
+            REQUIRE_THROWS(realm->update_schema(set_embedded(schema, "child_table", true), 2, nullptr));
+        }
+
+        SECTION("change table to embedded - no migration block") {
+            Schema schema = {
+                {"object",
+                 {
+                     {"value", PropertyType::Int},
+                 }},
+            };
+            auto realm = Realm::get_shared_realm(config);
+            realm->update_schema(schema, 1);
+            
+            REQUIRE_THROWS(realm->update_schema(set_embedded(schema, "object", true), 2, nullptr));
+        }
+
+        SECTION("change table to embedded - table has no backlinks") {
+            Schema schema = {
+                {"object",
+                 {
+                     {"value", PropertyType::Int},
                  }},
             };
             auto realm = Realm::get_shared_realm(config);
             realm->update_schema(schema, 1);
+            
+            REQUIRE_THROWS(realm->update_schema(set_embedded(schema, "object", true), 2,
+                                                 [](auto old_realm, auto new_realm, auto&) {}));
+        }
 
+        SECTION("change table to embedded - one incoming link per object") {
+            Schema schema = {
+                {"child_table",
+                 {
+                     {"value", PropertyType::Int},
+                 }},
+                {"parent_table",
+                 {
+                     {"child_property", PropertyType::Object | PropertyType::Nullable, "child_table"},
+                 }},
+            };
+            auto realm = Realm::get_shared_realm(config);
+            realm->update_schema(schema, 1);
             realm->begin_transaction();
-            auto target_table = ObjectStore::table_for_object_type(realm->read_group(), "target");
-            auto obj = target_table->create_object().get_key();
-            auto link_table = ObjectStore::table_for_object_type(realm->read_group(), "link");
-            link_table->create_object().set_all(obj);
-            link_table->create_object().set_all(obj);
+            auto child_table = ObjectStore::table_for_object_type(realm->read_group(), "child_table");
+            Obj child_object1 = child_table->create_object();
+            child_object1.set("value", 42);
+            Obj child_object2 = child_table->create_object();
+            child_object2.set("value", 43);
+            auto parent_table = ObjectStore::table_for_object_type(realm->read_group(), "parent_table");
+            auto child_object_key1 = child_object1.get_key();
+            auto child_object_key2 = child_object2.get_key();
+            parent_table->create_object().set_all(child_object_key1);
+            parent_table->create_object().set_all(child_object_key2);
             realm->commit_transaction();
+            REQUIRE(parent_table->size() == 2);
+            REQUIRE(child_table->size() == 2);
+
```

Comment:
suggest adding some checks around `child_table->is_embedded()` before and after the migration